app/environments/brassbirmingham/brassbirmingham/envs/classes/player.py

```diff
@@ -90,18 +104,21 @@ def decreaseLevel():
         for _ in range(levels):
             decreaseLevel()
 
+    # pass "money" object
     def canAffordBuildingIndustryResources(
-        self, buildLocation: BuildLocation, coalCost: int, ironCost: int
+        self, buildLocation: BuildLocation, coalCost: int, ironCost: int, money: int
     ) -> bool:
-        return (
+        #first check if that amount is available
+        return ((
             self.board.isCoalAvailableFromBuildings(buildLocation.town)
             or self.board.isCoalAvailableFromTradePosts(
-                buildLocation.town, coalCost, self.money
+                buildLocation.town, coalCost, money
             )
         ) and (
             self.board.isIronAvailableFromBuildings()
-            or self.board.isIronAvailableFromTradePosts(ironCost, self.money)
-        )
+            or self.board.isIronAvailableFromTradePosts(ironCost, money)
+        # if not, check if player can afford
+        )) or (self.board.priceForCoal(coalCost) + self.board.priceForIron(ironCost) <= money)
```

Comment:
Don't `isCoalAvailableFromTradePosts` and `isIronAvailableFromTradePosts` check against the cost? Technically the player may not be able to buy coal from the market if they're not connected

Reply:
Yep good catch